lib/Dialect/FIRRTL/Transforms/IMDeadCodeElim.cpp

```diff
@@ -116,9 +122,41 @@ struct IMDeadCodeElimPass : public IMDeadCodeElimBase<IMDeadCodeElimPass> {
   /// This keeps track of input ports that need to be kept if the associated
   /// instance is alive.
   DenseMap<InstanceOp, SmallVector<Value>> lazyLiveInputPorts;
+
+  /// A cache for a (inner)symbol lookp.
+  circt::hw::InnerRefNamespace *innerRefNamespace;
+  mlir::SymbolTable *symbolTable;
 };
 } // namespace
 
+void IMDeadCodeElimPass::markAlive(hw::HierPathOp hierPathOp) {
+  if (!liveHierPathOp.insert(hierPathOp).second)
+    return;
+  for (auto path : hierPathOp.getNamepathAttr())
+    if (auto innerRef = path.dyn_cast<hw::InnerRefAttr>()) {
+      auto op = innerRefNamespace->lookupOp(innerRef);
+      assert(op);
+      if (auto instance = dyn_cast<InstanceOp>(op))
+        // Mark the instance alive.
+        markAlive(instance);
+      // Otherwise, inner symbols are already marked alive.
+    }
+}
+
+void IMDeadCodeElimPass::markAlive(InstanceOp instance) {
+  if (!liveInstances.insert(instance).second)
+    return;
+  markBlockUndeletable(instance->getBlock());
+  if (instance.getInnerSym()) {
+    for (auto hierPathOp : instanceToHierpath[instance.getInnerRef()])
+      markAlive(hierPathOp);
+  }
+  // If the output port is alive, mark the instnace as alive. Propagate the
```

Comment:
nit: I know this comment already existed, but it seems like the `instnace` typo was carried over. If you're making any other fixes to this PR, maybe update this while you're at it?

Reply:
Yes, this comment is not accurate. I moved the function from the different place 😔